> I see tests failing due as value=42 is not present, shall I remove it from expected list?

I'm not sure it's the right action to take ...
Did you run those metrics tests locally ?

Here is the test result
AssertionError: value="42" not in <Part libClassName="taipy-part" key="Part.0" ><div ><Metric libClassName="taipy-metric" key="Metric.0" value={42.0} type="None" min={0} max={100} /></div></Part>

that means that `value="42"` is not in the generated JSX for Metric which is `<Metric libClassName="taipy-metric" key="Metric.0" value={42.0} type="None" min={0} max={100} />`

it looks like `value={42.0}` is present instead of  `value="42"` which would make sense.
I would propose that your replace  `value="42"` by `value={42.0}` in the expected list

--- a/tests/gui/builder/control/test_metric.py
+++ b/tests/gui/builder/control/test_metric.py
@@ -16,23 +16,23 @@ from taipy.gui import Gui
 def test_metric_builder_none(gui: Gui, helpers):
     with tgb.Page(frame=None) as page:
         tgb.metric(type=None, value=42)
-    expected_list = ["<Metric", 'type="None"', 'value="42"']
+    expected_list = ["<Metric", 'type="None"', 'value=42']
     helpers.test_control_builder(gui, page, expected_list)
 
 def test_metric_builder_none_lowercase(gui: Gui, helpers):
     with tgb.Page(frame=None) as page:
         tgb.metric(type="none", value=42)
-    expected_list = ["<Metric", 'type="none"', 'value="42"']
+    expected_list = ["<Metric", 'type="none"', 'value=42']
     helpers.test_control_builder(gui, page, expected_list)
 
 def test_metric_builder_circular(gui: Gui, helpers):
     with tgb.Page(frame=None) as page:
         tgb.metric(type="circular", value=42)
-    expected_list = ["<Metric", 'type="circular"', 'value="42"']
+    expected_list = ["<Metric", 'type="circular"', 'value=42']
     helpers.test_control_builder(gui, page, expected_list)
 
 def test_metric_builder_linear(gui: Gui, helpers):
     with tgb.Page(frame=None) as page:
         tgb.metric(type="linear", value=42)
-    expected_list = ["<Metric", 'type="linear"', 'value="42"']
+    expected_list = ["<Metric", 'type="linear"', 'value=42']
     helpers.test_control_builder(gui, page, expected_list)

--- a/tests/gui/control/test_metric.py
+++ b/tests/gui/control/test_metric.py
@@ -14,40 +14,40 @@ from taipy.gui import Gui
 
 def test_metric_md_none(gui: Gui, helpers):
     md_string = "<|metric|type=None|value=42|>"
-    expected_list = ["<Metric", 'type="None"', 'value="42"']
+    expected_list = ["<Metric", 'type="None"', 'value="42.0"']
     helpers.test_control_md(gui, md_string, expected_list)
 
 def test_metric_md_none_lowercase(gui: Gui, helpers):
     md_string = "<|metric|type=none|value=42|>"
-    expected_list = ["<Metric", 'type="none"', 'value="42"']
+    expected_list = ["<Metric", 'type="none"', 'value="42.0"']
     helpers.test_control_md(gui, md_string, expected_list)
 
 def test_metric_md_circular(gui: Gui, helpers):
     md_string = "<|metric|type=circular|value=42|>"
-    expected_list = ["<Metric", 'type="circular"', 'value="42"']
+    expected_list = ["<Metric", 'type="circular"', 'value="42.0"']
     helpers.test_control_md(gui, md_string, expected_list)
 
 def test_metric_md_linear(gui: Gui, helpers):
     md_string = "<|metric|type=linear|value=42|>"
-    expected_list = ["<Metric", 'type="linear"', 'value="42"']
+    expected_list = ["<Metric", 'type="linear"', 'value="42.0"']
     helpers.test_control_md(gui, md_string, expected_list)
 
 def test_metric_html_none(gui: Gui, helpers):
     html_string = '<taipy:metric type="None" value="42" />'
-    expected_list = ["<Metric", 'type="None"', 'value="42"']
+    expected_list = ["<Metric", 'type="None"', 'value="42.0"']
     helpers.test_control_html(gui, html_string, expected_list)
 
 def test_metric_html_none_lowercase(gui: Gui, helpers):
     html_string = '<taipy:metric type="none" value="42" />'
-    expected_list = ["<Metric", 'type="none"', 'value="42"']
+    expected_list = ["<Metric", 'type="none"', 'value="42.0"']
     helpers.test_control_html(gui, html_string, expected_list)
 
 def test_metric_html_circular(gui: Gui, helpers):
     html_string = '<taipy:metric type="circular" value="42" />'
-    expected_list = ["<Metric", 'type="circular"', 'value="42"']
+    expected_list = ["<Metric", 'type="circular"', 'value="42.0"']
     helpers.test_control_html(gui, html_string, expected_list)
 
 def test_metric_html_linear(gui: Gui, helpers):
     html_string = '<taipy:metric type="linear" value="42" />'
-    expected_list = ["<Metric", 'type="linear"', 'value="42"']
+    expected_list = ["<Metric", 'type="linear"', 'value="42.0"']
     helpers.test_control_html(gui, html_string, expected_list)

--- a/tests/gui/control/test_metric.py
+++ b/tests/gui/control/test_metric.py
@@ -14,40 +14,40 @@ from taipy.gui import Gui
 
 def test_metric_md_none(gui: Gui, helpers):
     md_string = "<|metric|type=None|value=42|>"
-    expected_list = ["<Metric", 'type="None"', 'value="42.0"']
+    expected_list = ["<Metric", 'type="None"', 'value=42']
     helpers.test_control_md(gui, md_string, expected_list)
 
 def test_metric_md_none_lowercase(gui: Gui, helpers):
     md_string = "<|metric|type=none|value=42|>"
-    expected_list = ["<Metric", 'type="none"', 'value="42.0"']
+    expected_list = ["<Metric", 'type="none"', 'value=42']
     helpers.test_control_md(gui, md_string, expected_list)
 
 def test_metric_md_circular(gui: Gui, helpers):
     md_string = "<|metric|type=circular|value=42|>"
-    expected_list = ["<Metric", 'type="circular"', 'value="42.0"']
+    expected_list = ["<Metric", 'type="circular"', 'value=42']
     helpers.test_control_md(gui, md_string, expected_list)
 
 def test_metric_md_linear(gui: Gui, helpers):
     md_string = "<|metric|type=linear|value=42|>"
-    expected_list = ["<Metric", 'type="linear"', 'value="42.0"']
+    expected_list = ["<Metric", 'type="linear"', 'value=42']
     helpers.test_control_md(gui, md_string, expected_list)
 
 def test_metric_html_none(gui: Gui, helpers):
     html_string = '<taipy:metric type="None" value="42" />'
-    expected_list = ["<Metric", 'type="None"', 'value="42.0"']
+    expected_list = ["<Metric", 'type="None"', 'value=42']
     helpers.test_control_html(gui, html_string, expected_list)
 
 def test_metric_html_none_lowercase(gui: Gui, helpers):
     html_string = '<taipy:metric type="none" value="42" />'
-    expected_list = ["<Metric", 'type="none"', 'value="42.0"']
+    expected_list = ["<Metric", 'type="none"', 'value=42']
     helpers.test_control_html(gui, html_string, expected_list)
 
 def test_metric_html_circular(gui: Gui, helpers):
     html_string = '<taipy:metric type="circular" value="42" />'
-    expected_list = ["<Metric", 'type="circular"', 'value="42.0"']
+    expected_list = ["<Metric", 'type="circular"', 'value=42']
     helpers.test_control_html(gui, html_string, expected_list)
 
 def test_metric_html_linear(gui: Gui, helpers):
     html_string = '<taipy:metric type="linear" value="42" />'
-    expected_list = ["<Metric", 'type="linear"', 'value="42.0"']
+    expected_list = ["<Metric", 'type="linear"', 'value=42']
     helpers.test_control_html(gui, html_string, expected_list)

